exam_1/exercise.py

```diff
@@ -0,0 +1,82 @@
+'''
+    1) Escribir una funcion que sacuda una matriz. Es decir, que las filas pares, las rote un elemento hacia la derecha, y las filas impares, hacia la izquierda. 
+    Esta función debe ser in-place, debe trabajar sobre la matriz recibida. La matriz esta representada como una lista de listas.
+    Ejemplo:
+    1 2 3 11    11 1 2 3
+    4 5 6 12 -> 5 6 12 4
+    7 8 9 13    13 7 8 9
+'''
+
+def sacudir(m):
+    ''' Función que recibe una matriz y mueve una posición los elementos.'''
+    for i,f in m:
+        if i%2 == 0:
+            #Inserto en la primera posición el elemento que saco del final
+            f.insert(0, f.pop())
+        else:
+            #Inserto en la última posición el elemento de la posición inicial
+            f.append(f.pop(0))
+    return m
```

Comment:
👎 no funciona
el enunciado dice que recibimos como matriz una lista de listas: [[1,2,3,11],[4,5,6,12],[7,8,9,13]]
por ende cuando iteremos la matriz vamos a estar recorriendo los elementos de m que son listas.
`for i,f in m:` esto no funciona

recordemos iteramos filas nada mas sin un indice, por lo que podriamos definir un contador para llevar el indice.
```
def sacudir(matriz):
    ''' Función que recibe una matriz y mueve una posición los elementos.'''
    indice = 0
    for fila in matriz:
        if indice % 2 == 0:
            #Inserto en la primera posición el elemento que saco del final
            fila.insert(0, fila.pop())
        else:
            #Inserto en la última posición el elemento de la posición inicial
            fila.append(fila.pop(0))
        indice += 1
    return matriz
```

PD: es importante que las variables sean descriptivas

Reply:
Me olvide el "enumerate", mala mía!